ArduPlane/commands_logic.cpp

```diff
@@ -943,6 +963,70 @@ void Plane::exit_mission_callback()
     }
 }
 
+bool Plane::verify_landing_vtol_approach(const AP_Mission::Mission_Command &cmd)
+{
+    switch (vtol_approach_s.approach_stage) {
+        case LOITER_TO_ALT:
+            if (verify_loiter_to_alt()) {
+                Vector3f wind = ahrs.wind_estimate();
+                vtol_approach_s.approach_direction = (degrees(atan2f(-wind.y, -wind.x)));
+                gcs().send_text(MAV_SEVERITY_INFO, "Selected an approach path of %f", (double)vtol_approach_s.approach_direction);
+                // select target approach direction
+                // select detransition distance (add in extra distance if the approach does not fit in the required space)
+                // validate turn
+                vtol_approach_s.approach_stage = WAIT_FOR_BREAKOUT;
+            }
+            break;
+        case WAIT_FOR_BREAKOUT:
+            {
+                const float breakout_direction = radians(wrap_180(vtol_approach_s.approach_direction + 270));
+
+                nav_controller->update_loiter(cmd.content.location, aparm.loiter_radius, cmd.content.location.flags.loiter_ccw ? -1 : 1);
+
+                // breakout when within 5 degrees of the opposite direction
+                if (fabsf(ahrs.yaw - breakout_direction) < radians(5.0f)) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Starting VTOL land approach path");
+                    vtol_approach_s.approach_stage = APPROACH_LINE;;
+                    set_next_WP(cmd.content.location);
+                    // fallthrough
+                } else {
+                    break;
+                }
+                FALLTHROUGH;
+            }
+        case APPROACH_LINE:
+            {
+                // project an apporach path
+                Location start, end;
+                memcpy(&start, &cmd.content.location, sizeof(Location));
+                memcpy(&end,   &cmd.content.location, sizeof(Location));
+
+                location_update(start, vtol_approach_s.approach_direction + 180, 1e3f);
+                location_update(end, vtol_approach_s.approach_direction, 1e3f);
+
+                nav_controller->update_waypoint(start, end);
+
+                // check if we should move on to the next waypoint
+                Location breakout_loc;
+                memcpy(&breakout_loc, &cmd.content.location, sizeof(Location));
+                location_update(breakout_loc, vtol_approach_s.approach_direction + 180, quadplane.stopping_distance());
+                if(location_passed_point(current_loc, start, breakout_loc)) {
+                    vtol_approach_s.approach_stage = VTOL_LANDING;
+                    quadplane.do_vtol_land(cmd);
+                    // fallthrough
```

Comment:
it looks like the fallthrough and the break both just end up with return false. What is the point of the if() ?

Reply:
It's a bit legacy, I can remove it, it was convenient to allow us to add code in the `VTOL_LANDING` case though, which was used for debugging. I'll go ahead and remove it at this point.

Reply:
And due to the above error found this is actually not changing, as there is now a functional difference.